src/indexer/log_merge_policy.rs

```diff
@@ -6,26 +6,40 @@ use std::f64;
 const DEFAULT_LEVEL_LOG_SIZE: f64 = 0.75;
 const DEFAULT_MIN_LAYER_SIZE: u32 = 10_000;
 const DEFAULT_MIN_MERGE_SIZE: usize = 8;
+const DEFAULT_MAX_MERGE_SIZE: usize = 5000;
 
 /// `LogMergePolicy` tries tries to merge segments that have a similar number of
 /// documents.
 #[derive(Debug, Clone)]
 pub struct LogMergePolicy {
     min_merge_size: usize,
+    max_merge_size: usize,
     min_layer_size: u32,
     level_log_size: f64,
+    consider_deletes: bool
 }
 
 impl LogMergePolicy {
     fn clip_min_size(&self, size: u32) -> u32 {
         cmp::max(self.min_layer_size, size)
     }
 
+    fn doc_count(&self, seg: &SegmentMeta) -> u32 {
+        if self.consider_deletes {
+            seg.num_docs() + seg.num_deleted_docs()
```

Comment:
that's `seg.max_doc()`... I am not sure this is a good heuristic here.
Can you remove the part of your PR that handles the deletes (and keep the `max_doc` thing) ?

I think the delete heuristic might require more discussion.

Reply:
Sounds good, removed it